src/porepy/numerics/ad/operators.py

```diff
@@ -1861,99 +1936,35 @@ def copy_common_sub_tags(self) -> None:
         return sum([v.size for v in self.sub_vars])
 
-    def previous_timestep(self) -> MixedDimensionalVariable:
-        """Return a representation of this mixed-dimensional variable on the previous
-        time step.
-
-        Raises:
-            ValueError:
-                If the variable is a representation of the previous iteration,
-                previously set by :meth:`~previous_iteration`.
-
-            NotImplementedError:
-                If the variable is already a representation of the previous time step.
-                Currently, we support creating only one previous time step.
-
-        Returns:
-            A representation of this merged variable on the previous time
-            iteration, with its ``prev_iter`` attribute set to ``True``.
-
-        """
-
-        new_subs = [var.previous_timestep() for var in self.sub_vars]
-        new_var = MixedDimensionalVariable(new_subs)
-        new_var.prev_time = True
-        # Assign self as the original variable.
-        new_var.original_variable = self
-        return new_var
-
-    def previous_iteration(self) -> MixedDimensionalVariable:
-        """Return a representation of this mixed-dimensional variable on the previous
-        iteration.
-
-        Raises:
-            ValueError:
-                If the variable is a representation of the previous time step,
-                previously set by :meth:`~previous_timestep`.
-
-            NotImplementedError:
-                If the variable is already a representation of the previous time
-                iteration. Currently, we support creating only one previous iteration.
-
-        Returns:
-            A representation of this merged variable on the previous
-            iteration, with its ``prev_iter`` attribute set to ``True``
-
-        """
-        new_subs = [var.previous_iteration() for var in self.sub_vars]
-        new_var = MixedDimensionalVariable(new_subs)
-        new_var.prev_iter = True
-        # Assign self as the original variable.
-        new_var.original_variable = self
-        return new_var
-
-    def copy(self) -> "MixedDimensionalVariable":
-        """Copy the mixed-dimensional variable.
-
-        Returns:
-            A shallow copy should be sufficient here; the attributes are not expected to
-            change.
-
-        """
-        return copy.deepcopy(self)
+    def parse(self, mdg: pp.MixedDimensionalGrid) -> Any:
+        """Returns a sequence of values stored for each variable in :attr:`sub_vars`."""
+        raise TypeError(
+            "Md-variables parsed on a md-grid without the equation system."
+            + " Use ``value(equation_system)`` instead."
+        )
 
-    def __repr__(self) -> str:
-        if self._no_variables:
-            return (
-                "Mixed-dimensional variable defined on an empty list of "
-                "subdomains or interfaces."
-            )
+    def previous_timestep(self, steps: int = 1) -> MixedDimensionalVariable:
```

Comment:
Custom prev time and iter behavior for MD variables, invoking the shift for all sub vars after super call